### PR TITLE
Add license

### PR DESCRIPTION
**Pull request by arzg**
Hi,

Thanks for the wonderful project! I was wondering if it would be possible to license it under MIT?

### CODE REVIEW

**Comment by justinmk** [comment]
Apache 2.0 would be best.

**Comment by arzg** [comment]
Sure. I’ll update now.

File: LICENSE.md [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+
+====
+
+Any parts of Neovim that were contributed under the Vim license are licensed
+under the Vim license unless the copyright holder gave permission to license
+those contributions under the Apache 2.0 license.
+
+The Vim license follows:
+
+VIM LICENSE
```
Comment (justinmk) [inline LICENSE.md]: no reason for this to be in the nvim-lsp repo.

File: LICENSE.md [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+====
+
+The above license applies to all parts of Neovim except (1) parts that were
+contributed under the Vim license and (2) externally maintained libraries.
+
+The externally maintained libraries used by Neovim are:
+
+  - Klib: a Generic Library in C. MIT/X11 license.
+  - libuv. Copyright Joyent, Inc. and other Node contributors. Node.js license.
+  - LuaJIT: a Just-In-Time Compiler for Lua. Copyright Mike Pall. MIT license.
```
Comment (justinmk) [inline LICENSE.md]: no reason for this whole section to be in the nvim-lsp repo.

I'd recommend using GitHub's license tools, IIRC they provide ready-made texts.

Comment (arzg) [reply LICENSE.md]: Yeah, sorry about that. I just copy-pasted Neovim’s license and just changed the header. I’ll keep the header, but change the text to GitHub’s copy of the Apache License.

**Comment by norcalli** [comment]
@justinmk I tend to just go with MIT license on personal projects, so I'll defer to your judgement on this one.

**Comment by justinmk** [comment]
Apache 2.0 is the typical license for Neovim projects, we should be consistent.

**Comment by norcalli** [comment]
@justinmk What I meant to say was "from my surface level experience with Apache 2.0, this looks good to merge, but I would feel more comfortable if you pressed the merge button" but I didn't phrase it properly.

**Pull request merged**